src/icons/ItemIcon.stories.tsx

```diff
@@ -40,63 +44,94 @@ export const Default: Story = {
   },
 };
 
+export const ImageWithStyle: Story = {
+  args: {
+    type: ItemType.FOLDER,
+    iconSrc: 'https://picsum.photos/200/100',
+    size: '100px',
+    sx: {
+      borderRadius: 2,
+    },
+  },
+};
+
 export const Image: Story = {
   args: {
     type: ItemType.S3_FILE,
     color: 'black',
-    extra: {
-      [ItemType.S3_FILE]: {
-        name: '',
-
-        mimetype: MimeTypes.Image.JPEG,
-      },
-    } as S3FileItemExtra,
+    mimetype: MimeTypes.Image.JPEG,
   },
 };
 
 export const Video: Story = {
   args: {
     type: ItemType.S3_FILE,
     color: 'black',
-    extra: {
-      [ItemType.S3_FILE]: {
-        mimetype: MimeTypes.Video.MP4,
-      },
-    } as S3FileItemExtra,
+    mimetype: MimeTypes.Video.MP4,
   },
 };
 
 export const Audio: Story = {
   args: {
     type: ItemType.S3_FILE,
     color: 'black',
-    extra: {
-      [ItemType.S3_FILE]: {
-        mimetype: MimeTypes.Audio.MP3,
-      },
-    } as S3FileItemExtra,
+    mimetype: MimeTypes.Audio.MP3,
   },
 };
 
 export const PDF: Story = {
   args: {
     type: ItemType.S3_FILE,
     color: 'black',
-    extra: {
-      [ItemType.S3_FILE]: {
-        mimetype: MimeTypes.PDF,
-      },
-    } as S3FileItemExtra,
+    mimetype: MimeTypes.PDF,
   },
 };
 
 export const ZIP: Story = {
   args: {
     type: ItemType.S3_FILE,
     color: 'black',
+    mimetype: MimeTypes.ZIP,
+  },
+};
+
+export const App: Story = {
+  args: {
+    type: ItemType.APP,
+    color: 'black',
+  },
+};
+
+export const Link: Story = {
+  args: {
+    type: ItemType.LINK,
+    color: 'black',
+  },
+};
+
+export const Shortcut: Story = {
+  args: {
+    type: ItemType.SHORTCUT,
+    color: 'black',
+  },
+};
+
+export const EtherPad: Story = {
+  args: {
+    type: ItemType.ETHERPAD,
+    color: 'black',
+  },
+};
+
+export const FancyImage: Story = {
+  args: {
+    type: ItemType.S3_FILE,
+    color: 'red',
+    sx: { fontSize: '3rem' },
     extra: {
       [ItemType.S3_FILE]: {
-        mimetype: MimeTypes.ZIP,
+        name: '',
+        mimetype: MimeTypes.Image.JPEG,
       },
     } as S3FileItemExtra,
```

Comment:
Is it to be sure it works with legacy code? Maybe add a comment that we can remove this test later.